Aztec/Classes/TextKit/TextView.swift

```diff
@@ -724,6 +767,56 @@ open class TextView: UITextView {
     }
 
 
+    /// Removes the List Attributes from a collection of attributes, whenever:
+    ///
+    ///     A. The caret is at the very end of the document (+ there is no selected text)
+    ///     B. There's a list!
+    ///     C. The previous character is a '\n'
+    ///
+    /// - Parameter attributes: Typing Attributes.
+    ///
+    /// - Returns: Updated Typing Attributes.
+    ///
+    private func ensureRemovalOfListAttribute(from attributes: [String: Any]) -> [String: Any] {
+        guard selectedRange.location == storage.length && selectedRange.length == 0 else {
+            return attributes
+        }
+
+        guard TextListFormatter.listsOfAnyKindPresent(in: attributes) else {
+            return attributes
+        }
+
+        let previousRange = NSRange(location: selectedRange.location - 1, length: 1)
+        let previousString = storage.safeSubstring(at: previousRange) ?? String(.newline)
+        guard previousString == String(.newline) else {
+            return attributes
+        }
+
+        var updatedAttributes = attributes
```

Comment:
From this line below, the problem is that we're not only removing the `textList` style, but we're actually resetting the full paragraph configuration to the default, which is not really desired.  For instance: we may not want to remove the blockquote style.

To avoid resetting the whole paragraph, you can use this code instead (or an equivalent code):

```swift
let orderedListFormatter = TextListFormatter(style: .ordered)
let unorderedListFormatter = TextListFormatter(style: .unordered)

if orderedListFormatter.present(in: attributes) {
    return orderedListFormatter.remove(from: attributes)
} else if unorderedListFormatter.present(in: attributes) {
    return unorderedListFormatter.remove(from: attributes)
} else {
    return attributes
}
```

PS: don't worry too much about the code being perfect.  This code will change ENORMOUSLY once we change to using a mechanism closer to what Apple uses for nested lists, but this should be a good starting point.